share/picongpu/tests/Synchrotron_plugin/lib/python/test/SynchrotronRadiation/validate.py

```diff
@@ -0,0 +1,174 @@
+import sys
+import openpmd_api as io
+import numpy as np
+import scipy.constants as const
+from scipy.integrate import quad
+from scipy.special import kv
+
+e = const.elementary_charge  # Elementary charge in C
+m_e = const.m_e  # Electron mass in kg
+c = const.c  # Speed of light in m/s
+hbar = const.hbar  # Reduced Planck constant in J/s
+
+miu0 = const.mu_0  # Vacuum permeability in H/m
+eps0 = const.epsilon_0  # Vacuum permittivity in F/m
+Es = m_e**2 * c**3 / e / hbar  # Schwinger limit in V/m
+
+
+def F1(z_q):
+    if z_q > 2.9e-6:
+        integral = quad(lambda x: kv(5 / 3, x), z_q, np.inf)[0]
+        return z_q * integral
+    else:
+        return 2.15 * z_q ** (1 / 3)
+
+
+def F2(z_q):
+    return z_q * kv(2 / 3, z_q)
+
+
+def analytical_Propability(delta, gamma, Heff, dt):
+    chi = gamma * Heff / Es
+    zq = 2 / (3 * chi) * delta / (1 - delta)
+    F1_result = F1(zq)
+    F2_result = F2(zq)
+    numerator = (
+        dt * e**2 * m_e * c * np.sqrt(3) * chi * (1 - delta) * (F1_result + 3 * delta * zq * chi / 2 * F2_result)
+    )
+    denominator = 2 * np.pi * gamma * delta * hbar**2 * eps0 * 4 * np.pi
+    return numerator / denominator
+
+
+def electron_energy(gamma):
+    return gamma * (m_e * c**2 * 6.2415e18)  # Energy of the electrons in eV
+
+
+# interpolate y1 to x0 and subtract y0 from y1
+def subtract_functions(x0, y0, x1, y1):
+    minX = max(np.min(x0), np.min(x1))
+    maxX = min(np.max(x0), np.max(x1))
+    #
+    x0 = [x for x in x0 if x >= minX and x <= maxX]
+    x1 = [x for x in x1 if x >= minX and x <= maxX]
+    y0 = [y for x, y in zip(x0, y0) if x >= minX and x <= maxX]
+    y1 = [y for x, y in zip(x1, y1) if x >= minX and x <= maxX]
+
+    y1 = np.interp(x0, x1, y1)  # interpolate y1 to the x0 values
+    return x0, np.abs(y0 - y1) / y0
+
+
+#    Returns energy in eV without the rest mass energy
+def momentum_to_energy(momentum, mass):
+    JtoeV = 6.242e18
```

Comment:
```suggestion
    JtoeV = const.elementary_charge**-1
```

Reply:
Could also be defined in a central file and then just used

Reply:
done